hazelcast/src/main/java/com/hazelcast/topic/impl/TotalOrderedTopicProxy.java

```diff
@@ -47,4 +62,53 @@ public void publish(@Nonnull  E message) {
         InternalCompletableFuture f = invokeOnPartition(operation);
         f.joinInternal();
     }
+
+    @Override
+    public InternalCompletableFuture<Void> publishAsync(@Nonnull E message) {
+        checkNotNull(message, NULL_MESSAGE_IS_NOT_ALLOWED);
+        Operation op = new PublishOperation(getName(), toData(message)).setPartitionId(partitionId);
+        return publishInternalAsync(op);
+    }
+
+    @Override
+    public void publishAll(@Nonnull Collection<? extends E> messages) {
+        checkNotNull(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
```

Comment:
You can just call `publishAllAsync` here.

Reply:
changed